requester/requester.go

```diff
@@ -7,18 +7,53 @@ import (
 	"encoding/base64"
 	"encoding/json"
 	"errors"
-	"io/ioutil"
+	"io"
 	"math/big"
 	"net/http"
 	"net/url"
 	"regexp"
 	"runtime"
+	"strconv"
 	"strings"
 	"time"
 
 	lightspark "github.com/lightsparkdev/go-sdk"
 )
 
+// RequestError indicates that a request to the Lightspark API failed.
+// It could be due to a service outage or a network error.
+// The request should be retried if RequestError is returned.
```

Comment:
@mgorven Do we ever return a 401 or 403 or are all http error codes deferred to graphql errors?

Reply:
I don't think so. This error is early returned before parsing graphql response.

Reply:
Right, but I'm suggesting that really only 5XX errors should be retries, likely not all 4XX errors if our backend is returning those ([which it seems to do](https://log.us-west-2.sparkinfra.net/_dashboards/goto/87d2c85541a970470f98915069e56dc1))

Reply:
Yeah that makes sense to me.

Reply:
We return 401 if an API token is provided but is invalid. It is also possible (but rare) for requests to fail with HTTP 500.